performance/scaling_data_ingestion.rst

```diff
@@ -3,79 +3,119 @@
+Second, these benchmark numbers show UPDATE throughput for standard Citus deployments. If you're on the Citus community edition, using statement-based replication, and you increased the default replication factor to 2, you're going to observe notably lower UPDATE throughputs. For this particular setting, Citus comes with additional configuration (citus.all_modifications_commutative) that may increase UPDATE ratios.
 
-For example, if your distributed table contains counters and all your DML queries are UPSERTs that add to the counters, then you can safely turn on citus.all_modifications_commutative since addition is commutative:
+Insert and Update: Throughput Checklist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you're running the above pgbench benchmarks on a moderately sized Citus cluster, you can generally expect 10K-50K INSERTs per second. This translates to approximately 1 to 4 billion INSERTs per day. If you aren't observing these throughputs numbers, remember the following checklist:
+
+* Check the network latency between your application and your database. High latencies will impact your write throughput.
+* Ingest data using concurrent threads. If the roundtrip latency during an INSERT is 4ms, you can process 250 INSERTs/second over one thread. If you run 100 concurrent threads, you will see your write throughput increase with the number of threads.
+* Check whether the nodes in your cluster have CPU or disk bottlenecks. Ingested data passes through the coordinator node, so check whether your coordinator is bottlenecked on CPU.
+* Avoid closing connections between INSERT statements. This avoids the overhead of connection setup.
+* Remember that column size will affect insert speed. Rows with big JSON blobs will take longer than those with small columns like integers.
+
+Insert and Update: Latency
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The benefit of running INSERT, INSERT .. ON CONFLICT, UPDATE, or DELETE commands is that changes are immediately visible to other queries. When you issue an INSERT or UPDATE command, the Citus coordinator node routes this command to related worker node(s). The coordinator node also keeps connections to the workers open within the same session, which means subsequent commands will see lower response times.
 
 ::
 
-    SET citus.all_modifications_commutative TO on;
-    INSERT INTO counters VALUES ('num_purchases', '2016-03-04', 1)
-    ON CONFLICT (c_key, c_date) DO UPDATE SET c_value = counters.c_value + 1;
+    -- Set up a distributed table that keeps account history information
+    CREATE TABLE pgbench_history (tid int, bid int, aid int, delta int, mtime timestamp, filler char(22));
```

Comment:
This example seems to have details that aren't related to showing a speedup from the one insert statement to the next. Like what is `filler` etc?

Maybe you're just going with a pg_bench theme here, but this section made me read more carefully because I thought it was trying to demonstrate something subtle.